[12.0][FIX] company_today: Execute every 15 min, remove doall

carmenbianca opened the pull request:
Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

## Description



## Odoo task (if applicable)



## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

carmenbianca committed:
[FIX] company_today: Execute every 15 min, remove doall

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

huguesdk commented on company_today/data/ir_cron.xml:
```diff
@@ -10,9 +10,19 @@ SPDX-License-Identifier: AGPL-3.0-or-later
         <field name="model_id" ref="model_res_company" />
         <field name="code">model.cron_update_today</field>
```
please update this branch (merge 12.0 or rebase), to include the fix that actually calls the code.

carmenbianca replied:
rebased without conflicts

huguesdk reviewed (commented):
i think that once per a few hours is enough (for our use case at least). for other use cases, once per hour is ok, i think. there is the (rather unlikely, but still possible) case where odoo is not running at the cron time, but in that case, today will only be off during one hour maximum. i think that this is a reasonable limitation (to put in the readme), compared to computing things over and over uselessly: even if the date does not change, all dependent computed values will be recomputed (since odoo does not check whether the value has changed or not), unless `.cron_update_today()` first checks whether the value needs to be updated.

carmenbianca commented:
> for other use cases, once per hour is ok, i think

I am inclined to think not. Nepal has an offset of UTC+5:45.

> unless .cron_update_today() first checks whether the value needs to be updated.

I knew I forgot something. I'll do that.

carmenbianca committed:
[FIX] company_today: Check whether date has changed before writing it

All fields are recomputed when the dependent field is written to, even
if the value of the dependent field did not change. By reducing the
amount of writes to the field, we reduce the amount of recomputation.

Signed-off-by: Carmen Bianca Bakker <carmen@coopiteasy.be>

robinkeunen commented:
> > for other use cases, once per hour is ok, i think
> 
> I am inclined to think not. Nepal has an offset of UTC+5:45.

Also the town of Eucla (UTC+08:45) at the border of West Australia (UTC+08:00) and Southern Australia (UTC+09:45). Apparently, it was more convenient to relay telegraph messages. Maintained for _retro-compatibility_ I guess...

robinkeunen commented:
@carmenbianca should we merge or close ?
